app/src/main/kotlin/com/intive/picover/profile/viewmodel/ProfileViewModel.kt

```diff
@@ -53,6 +55,21 @@ class ProfileViewModel @Inject constructor(
 				authRepository.updateUserAvatar(uri)
 			}.onSuccess {
 				_profile.value = Loaded(it)
+				_userName.value = it.name
+			}.onFailure {
+				// TODO NAN handle exception
+			}
+		}
+	}
+
+	fun updateName(name: String) {
+		viewModelScope.launch {
+			_profile.value = Loading
+			runCatching {
+				authRepository.updateUserName(name)
+			}.onSuccess {
+				_profile.value = Loaded(it)
+				_userName.value = it.name
```

Comment:
as we already discussed, please check later, is it possible to set user name automatically from updated profile value

Reply:
Done